paxos-commit, references, problem updates, todos

g4s8 opened the pull request:
Restructured text: moved some parts from solution to the problem to
exaplin it better;
Explained using paxos-commit;
Added references to bib file;
Added todos;

@stepanvalyavskiy can you please check this PR?

IlyaMoskva reviewed (commented):
@g4s8 ,please, take a look for my comments and probably fix "space" key on keyboard )

g4s8 commented on tech-report.tex:
```diff
@@ -13,77 +13,117 @@
+  % TODO: read, analyze and add to references.bib
+  %  - 3PC: https://ecommons.cornell.edu/handle/1813/6323 https://webee.technion.ac.il/~idish/Abstracts/jcss.html
+  %  - 2PC: check references section in https://en.wikipedia.org/wiki/Two-phase_commit_protocol
+  It is two-phase commit protocol (2PC), three-phase commit protocol (3PC) and paxos-commit~\cite{paxos-commit}.
```
@IlyaMoskva sure, it'll be added instead of `\todo{}`s

g4s8 commented on tech-report.tex:
```diff
@@ -13,77 +13,117 @@
-causes changes to be broadcast to all the Gitaly nodes in the cluster,
-but only the Gitaly nodes that vote in agreement with the primary node persist the changes to disk.
+causes changes to be broadcast to all the Gitaly nodes in the cluster~\todo{to all nodes or to all replicas of repository?},
+but only the Gitaly nodes that vote~\todo{we didn't explain voting previously} in agreement with the primary node persist the changes to disk \todo{this sentence is not very clear}.
```
@IlyaMoskva same here - we have a `\todo{}` for that

g4s8 commented:
@IlyaMoskva thanks, I've fixed document style according to tech-report requirements and fixed grammar mistakes with help of aspell checker (you may find it useful too: https://dbader.org/blog/spell-checking-latex-documents-with-aspell)

stepanvalyavskiy commented on tech-report.tex:
```diff
@@ -187,48 +219,37 @@ \section{Solution}
+  \item Create formal TLA+ specification for this algorithm
+  \item Design and implement transactions
+  \item Research for Paxos performance optimizations
+  \item Investigate self-healing mechanism~\todo{what is health-checking in context of atomic-commit?}
```
It's out of atomic-commit context and not a health-check.
Here I meant self-healing for entire Degitx. For ex. how to replace coordinator or node if they have crashed.
